lib/galaxy/webapps/galaxy/services/tools.py

```diff
@@ -98,7 +128,35 @@ def create_fetch(
         create_payload.update(files_payload)
         return self._create(trans, create_payload)
 
-    def _create(self, trans: ProvidesHistoryContext, payload, **kwd):
+    def execute(
+        self,
+        trans: ProvidesHistoryContext,
+        payload: ExecuteToolPayload,
+    ) -> ExecuteToolResponse:
+        tool_id = payload.tool_id
+        tool_uuid = str(payload.tool_uuid) if payload.tool_uuid else None
+        if tool_id in PROTECTED_TOOLS:
+            raise HTTPException(
+                status_code=400, detail=f"Cannot execute tool [{tool_id}] directly, must use alternative endpoint."
+            )
+        if tool_id is None and tool_uuid is None:
+            raise HTTPException(status_code=400, detail="Must specify a valid tool_id to use this endpoint.")
```

Comment:
You should always raise one of the galaxy exceptions. In this case, either `exceptions.RequestParameterInvalidException` or `exceptions.MessageException` have a look at https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/exceptions/__init__.py. This will ensure proper handling of the error in the client.